simple_api/object/actions.py

```diff
@@ -18,19 +19,30 @@ def set_name(self, name):
 
     def set_parent_class(self, cls):
         self.parent_class = cls
-        for field in {**self.parameters, **self.data}.values():
+        for field in self.parameters.values():
+            field.set_parent_class(cls)
+        for field in self.data.values():
             field.set_parent_class(cls)
         if self.return_value is not None:
             self.return_value.set_parent_class(cls)
 
 
 class Action(SetReferencesMixin, ToActionMixin):
-    def __init__(self, parameters=None, data=None, return_value=None, exec_fn=None, permissions=None, **kwargs):
+    def __init__(self, parameters=None, data=None, return_value=None, exec_fn=None, permissions=None, validators=None,
+                 **kwargs):
         self.parameters = parameters or {}
         self.data = data or {}
         self.return_value = return_value
         self.exec_fn = exec_fn
         self.permissions = permissions or ()
+        self.action_validators = validators or ()
+
+        # all field validators are read from argument and restructured into {"parameter": (validator1, validator2...)}
+        self.parameters_validators = {param_name: param.validators or ()
+                                      for param_name, param in self.parameters.items()}
+        self.data_validators = {field_name: field.validators or ()
+                                for field_name, field in self.data.items()}
```

Comment:
`or ()` je zbytecne, protoze v konstruktoru `Type` mas `ensure_tuple`, takze to bude vzdycky tuple, at uz prazdny, nebo ne